src/ruuvitag_ble/df3_decoder.py

```diff
@@ -24,13 +24,20 @@ def humidity_percentage(self) -> float | None:
 
     @property
     def temperature_celsius(self) -> float | None:
-        return round(self.data[2] + self.data[3] / 100.0, 2)
+        if self.data[3] > 99:
+            return None
+        fraction_part = round(self.data[3] / 100.0, 2)
+
+        # Ruuvi uses the MSB as the sign bit here
+        if self.data[2] & 0x80:
+            integer_part = -(self.data[2] & 0x7F)
+        else:
+            integer_part = self.data[2] & 0x7F
+
+        return integer_part + fraction_part
```

Comment:
```suggestion
        integer_part = self.data[2] & 0x7F
        # Ruuvi uses the MSB as the sign bit here
        if self.data[2] & 0x80:
            integer_part = -integer_part
        
        return integer_part + fraction_part
```

would be simpler... but should the fractional part also be negated, i.e.

```suggestion
        integer_part = self.data[2] & 0x7F
        # Ruuvi uses the MSB as the sign bit here
        sign = -1 if self.data[2] & 0x80 else +1
        return sign * (integer_part + fraction_part)
```